src/components/pages/TransactionHistoryPage.js

```diff
@@ -1,34 +1,106 @@
   const [page, setPage] = useState(1);
+  // State variables used to access algoquant SDK API and display/ keep state of user data from database
+  const algoquantApi = useContext(AlgoquantApiContext);
+  const [lastKey, setLastKey] = useState(null);
+  const [pagesSeen, setPagesSeen] = useState(0);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const fetchTrades = useCallback(
+    (FETCH_AMOUNT) => {
+      const historyBuffer = [];
+      if (!lastPage && pagesSeen < page) {
+        if (algoquantApi.token) {
+          algoquantApi
+            .getTrades(FETCH_AMOUNT, lastKey)
+            .then((resp) => {
+              setLastKey(null);
+              setPagesSeen(pagesSeen + 1);
+              if (resp.data.LastEvaluatedKey === undefined && page !== 1) {
+                setLastPage(true);
+              }
+              if (resp.data.LastEvaluatedKey !== undefined) {
+                setLastKey({
+                  timestamp: resp.data.LastEvaluatedKey.timestamp,
+                  user_id: resp.data.LastEvaluatedKey.user_id,
+                });
+              }
+              for (let i = 0; i < resp.data.Count; i++) {
+                let timestamp = new Date(
+                  parseInt(resp.data.Items[i].timestamp)
+                );
+                historyBuffer.push({
+                  jobName: resp.data.Items[i].job_name,
+                  buyOrSell: resp.data.Items[i].side === "B" ? "Buy" : "Sell",
+                  stockTicker: resp.data.Items[i].symbol,
+                  shares: resp.data.Items[i].qty,
+                  amount: formatter.format(resp.data.Items[i].avg_price),
+                  date: timestamp.toLocaleString(),
+                });
+              }
+
+              setHistory(history.concat(historyBuffer));
+              setIsLoading(false);
+            })
+            .catch((err) => {
+              // TODO: Need to implement better error handling
+              console.log(err);
+            });
+        }
+      }
+    },
+    [algoquantApi, page, history, lastPage, pagesSeen, lastKey]
+  );
 
   useEffect(() => {
     const newTransactions = [];
-    for (let i = (page - 1) * 20; i < page * 20; i++) {
-      newTransactions.push({
-        jobName: `Job ${i + 1}`,
-        buyOrSell: i & (2 === 0) ? "Buy" : "Sell",
-        stockTicker: "APPL",
-        shares: i + 1,
-        amount: i * 100,
-        date: "01/01/2022 8:55PM",
-      });
+    let itemCounter = 0;
+    fetchTrades(FETCH_AMOUNT);
+    //  this is whats gonna handle what shows on screen
+    if (history.length !== 0) {
+      for (let i = (page - 1) * FETCH_AMOUNT; i < history.length; i++) {
+        if (itemCounter === FETCH_AMOUNT) break;
+        newTransactions.push(history[i]);
+        itemCounter++;
+      }
     }
     setTransactions(newTransactions);
-  }, [page]);
+  }, [history, fetchTrades, page]);
 
   const handleNextClick = () => {
+    if (!lastPage && pagesSeen <= page) {
```

Comment:
Do you need to check for `!lastPage` ? This button shouldn't be available if you are on the last page.

Reply:
removed